uwblueprint/humane-society
# S24/daniel/create password page

## Notion ticket link

[Set New Password Page](https://www.notion.so/uwblueprintexecs/Set-New-Password-Page-12210f3fb1dc80deb18eed180d09c23d?pvs=4)

## Implementation description
* Made the CreatePasswordPage which displays the user's email and two password fields, with validation.
* If viewed on a phone in landscape, it becomes scrollable
* Created responsive components for common components.
* Changing password in create password page changes your password. Next time you log in, the updated password is used.



## Steps to test
1. Go to localhost:3000/create-password
2. Password fields should throw errors if they don't match or if they're less than 8 characters. Otherwise, it opens the confirmation modal.


## What should reviewers focus on?
* Should the route be a PrivateRoute with allowedRoles={AuthConstants.ALL_ROLES}?
* Should I add any other validators? Also, is there a better way I should implement form validation?
* Any coding style/visual things that look off
* Is there a better way to apply different styles based on device orientation?

## Checklist
- [X ] My PR name is descriptive and in imperative tense
- [ X] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ X] I have run the appropriate linter(s)
- [ X] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

## Mobile:
<img width="383" alt="image" src="https://github.com/user-attachments/assets/6900934e-6b12-4500-8c4b-029c740f51c2">

<img width="684" alt="image" src="https://github.com/user-attachments/assets/921d4c81-ef9f-43b1-ba7c-840ac0842615">
<img width="684" alt="image" src="https://github.com/user-attachments/assets/0723f5c5-1605-4ae4-8640-30ea6a935092">

## Tablet:
<img width="773" alt="image" src="https://github.com/user-attachments/assets/d0a20390-9809-4d29-ae27-dde95eef5c3d">
<img width="468" alt="image" src="https://github.com/user-attachments/assets/1cd553a5-42ac-4a9d-992e-e42ae39ddc39">
